extra-modules/extra-phishing/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/phishing/PhishingExtension.kt

```diff
@@ -278,19 +282,24 @@ class PhishingExtension(private val settings: ExtPhishingBuilder) : Extension()
         val domains: MutableSet<String> = mutableSetOf()
 
         for (match in settings.urlRegex.findAll(content)) {
-            val found = match.groups[1]!!.value.trim('/')
+            val found = match.groups[1]?.value?.trim('/')
+
+            if (found == null) {
+                continue
+            }
+
             var domain = found
 
             if ("/" in domain) {
                 domain = domain
                     .split("/", limit = 2)
-                    .first()
-                    .lowercase()
+                    .firstOrNull()
+                    ?.lowercase()
             }
 
-            domain = domain.filter { it.isLetterOrDigit() || it in "-+&@#%?=~_|!:,.;" }
+            domain = domain?.filter { it.isLetterOrDigit() || it in "-+&@#%?=~_|!:,.;" }
 
-            if (domain in domainCache) {
```

Comment:
If the domain is null, we should have `continue`d or thrown earlier, given the logic you've written - it doesn't make sense to try to continue processing if there's no domain